ppci/lang/c/parser.py

```diff
@@ -876,9 +876,10 @@ def parse_switch_statement(self):
         self.consume("(")
         expression = self.parse_expression()
         self.consume(")")
-        self.semantics.on_switch_enter(expression)
+        # expression type can be changed by integer promotion
+        prom_expression = self.semantics.on_switch_enter(expression)
```

Comment:
I would simply rename this to `expression` again. I was confused that it was a promised expression.

Reply:
No problem. In fact "prom_expression" was for "promoted_expression"
I used a new name to highlight the fact that the object it references can be different from the object referenced by "expression" (then I added the comment to clarify)